### DISCUSSION TITLE
Unable to connect AD (TLS) server on Linux machine

### DISCUSSION THREAD

**Discussion by Sheetal1790** [dsbenghe/Novell.Directory.Ldap.NETStandard#230] [General]
I have a web application that uses Novell.Directory.Ldap.NETStandard library to do Ldap search against Active Directory. 
On Windows machine I am able to connect and fetch details from both AD (TLS) and AD (Non-TLS) server. But when the application is deployed on Linux machine I am only able to connect to AD (Non-TLS)  server, while on AD (TLS) server I am getting below error:

**Below issue is displayed on Linux machine:**
_Unable to read data from the transport connection: Connection reset by peer._

**Below issue displayed on AD server:**
_An TLS 1.2 connection request was received from a remote client application, 
but none of the cipher suites supported by the client application are supported by the server. 
The SSL connection request has failed._

Is there any way to resolve this issue. Can anyone help on this?

**Comment by dsbenghe** [comment]
Typical error of incompatible ciphers between client and server. The Linux host may disable some ciphers that are considered insecure.  
Nothing to do with the library per se; you can try some of the Linux ldap command line tools and you should get the same issue.

**Comment by Sheetal1790** [comment]
Thanks for taking a look into this. It helped.

**Reply by alberk8** [reply]
What was the issue? How did you resolved it? Thank you.
